src/routes/role-router.js

```diff
@@ -3,21 +3,44 @@
 const express = require('express');
 const router = express.Router();
 
+const Roles = require('../models/roles-model.js');
+const roles = new Roles();
+
+const auth = require('../middleware/auth.js');
+const err401 = require('../middleware/403.js');
+const err403 = require('../middleware/403.js');
+
 // TODO: Swagger Comments
 // Visible by all clients
-router.get('/public', (req, res, next) => {
+router.get('/public', err401, (req, res, next) => {
   res.status(200).json({ valid: true });
 });
 
 // === TODO: Define all the routes below ======
 
+router.use(auth);
+
 // TODO: Swagger Comments
 // Visible by logged in clients
-router.get('/hidden', (req, res, next) => {});
+router.get('/hidden', auth, (req, res, next) => {
+  if (req.user&& req.user._id){
+    res.status(200).json({ valid: true });
+  } else next ('Unable to find user');
+}, err403);
 
 // TODO: Swagger Comments
 // Visible by roles that have the "read" capability
-router.get('/read-only', (req, res, next) => {});
+router.get('/read-only', auth, err401, async (req, res, next) => {
+  if (req.user&& req.user._id){
+    let role = req.user.role;
```

Comment:
Once you finish writing the `users.methods.can()` in `user-schema.js` you'll then use it here.  **hint something like: `if(req.user.can('read')...`